File: Back to Basics Approach 3rd Edition/CodeStepByStep_java/javascript/recursion/factorial.js
/**
 * https://www.codestepbystep.com/problem/view/javascript/recursion/factorial
 *
 * @author Alex Golub
 * @since 03-May-21, 7:38 PM
 */
function factorial(n) {
  if (n < 0) {
    throw 'n must be non-negative';
  } else if (n <= 1) {
    return 1;
  }
  return n * factorial(n - 1);
}

File: Back to Basics Approach 3rd Edition/CodeStepByStep_java/recursion/factorial.java
package CodeStepByStep.java.recursion;

/**
 * https://www.codestepbystep.com/problem/view/java/recursion/factorial
 *
 * @author Alex Golub
 * @since 03-May-21, 7:37 PM
 */
class factorial {
  public static int factorial(int n) {
    if (n <= 1) return 1;
    return n * factorial(n - 1);
  }
}
